subprojects/testing-jvm/src/test/groovy/org/gradle/api/internal/tasks/testing/detection/TestClasspathFactoryTest.groovy

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.detection
+
+
+import org.gradle.api.internal.classpath.Module
+import org.gradle.api.internal.classpath.ModuleRegistry
+import org.gradle.api.internal.tasks.testing.TestFramework
+import org.gradle.internal.classpath.ClassPath
+import spock.lang.Specification
+
+/**
+ * Tests {@link TestClasspathFactory}.
+ */
+class TestClasspathFactoryTest extends Specification {
+
+    // The number of internal and external implementation jars loaded from the distribution regardless of framework.
+    private static final int NUM_INTERNAL_JARS = 19
+    private static final int NUM_EXTERNAL_JARS = 6
+
+    ModuleRegistry moduleRegistry = Mock(ModuleRegistry) {
+        getModule(_) >> { module(it[0], false) }
+        getExternalModule(_) >> { module(it[0], true) }
+    }
```

Comment:
💭 This could be static setup in `setupSpec()` rather than done every test method.

Reply:
I'm going to leave this for now. We very rarely use this pattern elsewhere and I'd like to remain consistent. I don't think the performance implications of re-instantiating these mocks are significant in the grand scheme of things. If you think this pattern is useful, lets have a separate discussion about it.

Reply:
👍 No need, just pointing it out.